File: implementations/react-native/design-patterns-rn/src/patterns/behavioral/memento/originator.ts
import { EditorMemento } from './memento';

export class EditorOriginator {
  private text = '';

  setText(t: string) { this.text = t; }
  getText() { return this.text; }

  createMemento(): EditorMemento {
    return new EditorMemento(this.text);
  }

  setMemento(m: EditorMemento) {
    this.text = m.getState().text;
  }
}

File: implementations/flutter/design_patterns_flutter/lib/patterns/behavioral/memento/originator.dart
import 'memento.dart';

class EditorOriginator {
  String _text = '';

  void setText(String t) => _text = t;
  String getText() => _text;

  EditorMemento createMemento() => EditorMemento(_text);
  void setMemento(EditorMemento m) {
    final s = m.getState();
    _text = (s['text'] as String?) ?? '';
  }
}
